state/cleanup.go

```diff
@@ -1130,26 +1131,26 @@ func (st *State) forceRemoveUnitStorageAttachments(unit *Unit) error {
 	return nil
 }
 
-func (st *State) cleanupForceRemoveUnit(ctx context.Context, store objectstore.ObjectStore, applicationService ApplicationService, unitId string, cleanupArgs []bson.Raw) error {
+func (st *State) cleanupForceRemoveUnit(ctx context.Context, store objectstore.ObjectStore, applicationService ApplicationService, unitName string, cleanupArgs []bson.Raw) error {
 	var maxWait time.Duration
 	if n := len(cleanupArgs); n != 1 {
 		return errors.Errorf("expected 1 argument, got %d", n)
 	}
 	if err := cleanupArgs[0].Unmarshal(&maxWait); err != nil {
 		return errors.Annotate(err, "unmarshalling cleanup arg 'maxWait'")
 	}
-	unit, err := st.Unit(unitId)
+	unit, err := st.Unit(unitName)
 	if errors.Is(err, errors.NotFound) {
-		logger.Debugf("no need to force remove unit %q", unitId)
+		logger.Debugf("no need to force remove unit %q", unitName)
 		return nil
 	} else if err != nil {
 		return errors.Trace(err)
 	}
 	opErrs, err := unit.RemoveWithForce(store, true, maxWait)
 	if len(opErrs) != 0 {
-		logger.Warningf("errors encountered force-removing unit %q: %v", unitId, opErrs)
+		logger.Warningf("errors encountered force-removing unit %q: %v", unitName, opErrs)
 	} else {
-		err = applicationService.DeleteUnit(ctx, unitId)
+		err = applicationService.DeleteUnit(ctx, coreunit.Name(unitName))
```

Comment:
**todo:** If it is safe to not une NewName here (and validate), add a comment that state it.